src/data/workspaces/workspace-all-query.ts

```diff
@@ -0,0 +1,37 @@
+import { apiPaths, createApiPath } from "../api";
+import { FetchError } from "@/lib/fetch-error";
+import { notFound } from "@/lib/not-found-error";
+import { fetcher } from "../fetch";
+import { useQuery, UseQueryOptions } from "@tanstack/react-query";
+
+export async function getWorkSpaceDetail(workspaceName: string) {
+	const url = createApiPath(apiPaths.workspaces.detail(workspaceName));
+	const res = await fetcher(url, {
+		method: "GET",
+		headers: {
+			"Content-Type": "application/json"
+		}
+	});
+
+	if (res.status === 404) notFound();
+
+	if (!res.ok) {
+		throw new FetchError({
+			message: "Error fetching secret details",
+			status: res.status,
+			statusText: res.statusText
+		});
+	}
+	return res.json();
+}
+
+export function useGetWorkSpaceDetail(
+	workspaceName: string,
+	options?: Omit<UseQueryOptions<any, unknown, any>, "queryFn" | "queryKey">
+) {
+	return useQuery<any, unknown, any>({
+		queryFn: () => getWorkSpaceDetail(workspaceName),
+		queryKey: ["workspaceDetail", workspaceName],
```

Comment:
@Aagya-Sharma - is it okay to use `workspaceName` here? or does it make sense to use an `id` field that doesnt change. What would happen if the workspace is renamed?

Reply:
what they mentioned was they dont really use workspaces now they are trying to change the flow as they are changing the concept of having multiple workspaces so they suggested me to use workspace name: which is 'default' always and then access the workspace id of it required for the creation of new secret